### ISSUE TITLE
Add hover state to Footer Social Media Icons

### ISSUE THREAD

**Issue by blilianyu** [NCIOCPL/ncids#2113]
### Issue description
Social media icons in the footer should have a hover state for accessibility.

### What's the expected change?
- Social media icons will have a hover state color `accent-warm-dark` #DDAA01

### What's the current functionality?
- There's no hover state

### What's the updated acceptance criteria?
- When a user hovers over a social media icon that is placed in an NCIDS Footer, then the social media icon will have the hover state color listed above.
- The NCIDS Documentation Site, NCIDS Storybook, and NCIDS Example Site will all display the update to the social media icon having a hover state
  - On the NCIDS Documentation Site, the icons in the footer of every page will have the update to the social media icons, as well as the Footer component page, with the appropriate updates to the code snippets and previews
  - On the NCIDS Storybook, the 6 storybook pages with Footer examples/variants will be updated to include the hover state
  - On the Example Site, all pages on the site have a footer with icons that need to be updated with the hover state except for https://designsystem.cancer.gov/example-site/example-pages/accordion.html
